app/controllers/billboards_controller.rb

```diff
@@ -0,0 +1,91 @@
+class BillboardsController < ApplicationController
+	
+  @@show_free = false
+
+
+
+	before_action :authenticate_user!, :except => [:show, :index]
+
+	
+	def index
+		#@billboards = Billboard.all
+		if @@show_free == true
+			@billboards = Billboard.all.where('id NOT IN (?)', Rent.where(:is_active? == true).where('start <= ?', DateTime.now.beginning_of_day).select(:board_id))
+		else
+		  @billboards = Billboard.all
+		end
+	end
+
+	def free
+		redirect_to billboards_path
+	  @billboards = Billboard.all.where('id NOT IN (?)', Rent.where(:is_active? == true).where('start <= ?', DateTime.now.beginning_of_day).select(:board_id))
+
+	end
+
+	def show
+		@billboard = Billboard.find(params[:id])
+		@request = Request.new
+		@requests = @billboard.requests
+		
+	end
+
+	def new
+		auth_admin
+		#if current_user.admin
+			@billboard = Billboard.new
+		#session[:billboard_params] ||= {}
+
+		#else
+		#	redirect_to root_path
+		#end
+	end
+
+	def create
+
+	auth_admin
```

Comment:
If you call something multiple times in different method consider extracting it into `before_action`